main.tf

```diff
@@ -11,6 +11,53 @@ resource "ibm_resource_instance" "scc_instance" {
   tags              = var.resource_tags
 }
 
+data "ibm_scc_provider_type" "scc_provider_type" {
+  count              = var.wp_instance_crn != null ? 1 : 0
+  provider_type_name = var.provider_type_name
```

Comment:
FYI, when https://github.com/IBM-Cloud/terraform-provider-ibm/issues/5208 is implemented, this module wont have a `provider_type_name` variable, instead you will hard code "Security and Compliance Workload Protection" here nd do data lookup by name

Reply:
it won't take anything, instead we just get a list of every provider type by the looks of it https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5179/files#diff-268bf1a5e7374e3609b776536171ae795dc6c12c9034ff087d169a59635cb3d6R19

Reply:
Once we can lookup the "Security and Compliance Workload Protection" by name and get its ID, we should be good here